internal/catalogmetadata/client/client.go

```diff
@@ -31,14 +51,42 @@ func (c *Client) Bundles(ctx context.Context) ([]*catalogmetadata.Bundle, error)
 		return nil, err
 	}
 	for _, catalog := range catalogList.Items {
-		channels, err := fetchCatalogMetadata[catalogmetadata.Channel](ctx, c.cl, catalog.Name, declcfg.SchemaChannel)
+		// if the catalog has not been successfully unpacked, skip it
+		if !meta.IsStatusConditionPresentAndEqual(catalog.Status.Conditions, catalogd.TypeUnpacked, metav1.ConditionTrue) {
+			continue
+		}
```

Comment:
Follow-up: we probably should not ignore unavailable catalogs. If we do, we introduce non-determinism to resolution.

If a `Catalog` that is needed for resolution is unavailable (that's all `Catalogs` right now because there's no way to disable catalogs globally or for a particular `Operator`), I think we should fail resolution, report the error, and then requeue when the Catalog changes state.

Reply:
That means that all `Operator` installs have a dependency on all `Catalog` resources being available. If someone creates a `Catalog` resource with an improper image (i.e not an actual catalog image) that would result in being unable to install any `Operator`s until that is remedied which feels like a bad UX IMO. The way I am viewing it is that at resolution time we only care about resolving based on the `Catalog`s that are available and have been successfully unpacked. If none of them provide what we need, fail resolution. The exception here is if a user specifies that an `Operator` should _only_ be installed via a specific `Catalog` (which I'm not sure if that is currently possible). 

Definitely agree that this is a follow-up item and needs further discussion. I created #428 to capture this.